.github/workflows/test.yml

```diff
@@ -42,7 +42,7 @@ jobs:
           node <<'SCRIPT'
             process.env["INPUT_TARANTOOL-VERSION"] = "${{ matrix.tarantool }}"
             process.env["INPUT_NIGHTLY-BUILD"] = "${{ matrix.nightly }}"
-            require("./dist/main").latest_version().then(v => {
+            require("./dist/main").latest_version("${{ matrix.tarantool }}").then(v => {
```

Comment:
Is `.github/actions/latest-version/action.yml` affected as well?

Reply:
Got it. `2.10` is not a latest minor release of the series-2. `2.11` is the latest one. So, we need to add the `version_prefix` argument to the `latest_version` call here.

Regarding the `latest_version` action: its argument is called `tarantool-series` and it seems OK to pass it to `inputs.tarantool-version` and don't pass it to the `version_prefix` argument. If we need a version filtering from the action, its argument should be renamed or another argument is added. I'm OK to leave it without changes.

Reply:
I'm not sure `tarantool-version` works as intended right now. We set the `INPUT_TARANTOOL-VERSION` here, but it doesn't seem to be used in the code we call:

Reply:
It is used in the `construct_base_url()`, which is called from here. In fact, it affects only release series (i. e. which repostiory is used), but doesn't perform version filtering within the repository. So, it doesn't distinguish 2.10 and 2.11, because they're in the same `series-2` repository.

Reply:
How about the following patch? It will eliminate the necessity to provide argument
```diff
--- a/src/main.ts
+++ b/src/main.ts
@@ -165,7 +165,9 @@ async function available_versions(
     })
 }

-export async function latest_version(version_prefix: string): Promise<string> {
+export async function latest_version(
+  version_prefix: string = tarantool_version
+): Promise<string> {
   return available_versions(version_prefix).then(versions => {
     let max = ''
     versions.forEach(v => {
```

Reply:
I have no objections.

Alternatively we can just strip the argument. It seems, it interferes with the global `tarantool_version` in a confusing way.

<details>
<summary>Patch (untested)</summary>

```diff
diff --git a/src/main.ts b/src/main.ts
index 859c656..70b6a46 100644
--- a/src/main.ts
+++ b/src/main.ts
@@ -115,14 +115,12 @@ function construct_base_url(): string {
   )
 }
 
-async function available_versions(
-  version_prefix: string
-): Promise<Array<string>> {
+async function available_versions(): Promise<Array<string>> {
   const baseUrl = construct_base_url()
   const repo = baseUrl + '/ubuntu/dists/' + (await lsb_release())
 
   // Don't return 1.10.10, when the version prefix is 1.10.1.
-  const prefix = version_prefix ? version_prefix + '.' : ''
+  const prefix = tarantool_version + '.'
 
   return http_get(`${repo}/main/binary-amd64/Packages`)
     .then(response => {
@@ -140,7 +138,7 @@ async function available_versions(
           const match = paragraph.match(/^Version: (.+)$/m)
           if (match) {
             let v = match[1]
-            if (!prefix || v.startsWith(prefix)) {
+            if (v.startsWith(prefix)) {
               versions.push(v)
             }
           }
@@ -149,8 +147,8 @@ async function available_versions(
     })
 }
 
-export async function latest_version(version_prefix: string): Promise<string> {
-  return available_versions(version_prefix).then(versions => {
+export async function latest_version(): Promise<string> {
+  return available_versions().then(versions => {
     let max = ''
     versions.forEach(v => {
       max = semver_max(max, v)
@@ -166,7 +164,7 @@ async function run_linux(): Promise<void> {
     const baseUrl = construct_base_url()
 
     core.startGroup(`Checking latest tarantool ${tarantool_version} version`)
-    const version = await latest_version(tarantool_version)
+    const version = await latest_version()
     core.info(`${version}`)
     core.endGroup()
     if (core.getInput('cache-key')) {
```
</details>

Reply:
I don't see a reason to strip it:
1) We may in future need to pass it explicitly
2) The diff is smaller with default value of the argument)

Let's leave it as is.

Reply:
OK, no problems.